### PR TITLE
[JBPM-9780] Accept all kafka properties

### PR DESCRIPTION
**Pull request by fjtirado**
**JIRA**: 

[link](https://issues.redhat.com/browse/JBPM-9780)

### CODE REVIEW

**Review by elguardian — APPROVED** [review]

File: kie-server-parent/kie-server-services/kie-server-services-kafka/src/test/java/org/kie/server/services/jbpm/kafka/KafkaServerExtensionTest.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        System.setProperty(KAFKA_EXTENSION_PREFIX + ProducerConfig.BATCH_SIZE_CONFIG, "1000");
+        System.setProperty(KAFKA_EXTENSION_PREFIX + "randomProperty", "JOJOJO");
+        try {
+            Map<String, Object> producerProperties = ext.getProducerProperties();
+            Map<String, Object> consumerProperties = ext.getConsumerProperties();
+            ext.initProperties();
+            assertEquals(1, producerProperties.size());
+            assertTrue(producerProperties.containsKey(ProducerConfig.BATCH_SIZE_CONFIG));
+            assertEquals(1, consumerProperties.size());
+            assertTrue(consumerProperties.containsKey(ConsumerConfig.ALLOW_AUTO_CREATE_TOPICS_CONFIG));
```
Comment (gmunozfe) [inline kie-server-parent/kie-server-services/kie-server-services-kafka/src/test/java/org/kie/server/services/jbpm/kafka/KafkaServerExtensionTest.java:29]: You could also assert the value of these properties. 
"randomProperty" is not checked that it's not included in any of the properties.

File: kie-server-parent/kie-server-services/kie-server-services-kafka/src/main/java/org/kie/server/services/jbpm/kafka/KafkaServerProducer.java [hunk 2]
```diff
@@ -75,7 +75,12 @@ private void sendEvent(ProcessInstance processInstance,
                            String name,
                            Object value) {
         if (producerReady.compareAndSet(false, true)) {
-            producer = producerSupplier.get();
+            try {
+                producer = producerSupplier.get();
+            }
+            catch (Exception ex) {
+                producerReady.set(false);
```
Comment (gmunozfe) [inline kie-server-parent/kie-server-services/kie-server-services-kafka/src/main/java/org/kie/server/services/jbpm/kafka/KafkaServerProducer.java:82]: Shouldn't the exception be thrown again? Or at least, return from this method.
As the producer is not ready will fail in the send call.

File: kie-server-parent/kie-server-services/kie-server-services-kafka/src/main/java/org/kie/server/services/jbpm/kafka/KafkaServerExtension.java [hunk 3]
```diff
@@ -260,4 +244,26 @@ public void disposeContainer(String id, KieContainerInstance kieContainerInstanc
...
+
+    protected final void initProperties() {
+        for (Map.Entry<Object, Object> entry : System.getProperties().entrySet()) {
+            String key = entry.getKey().toString();
+            if (key.startsWith(KAFKA_EXTENSION_PREFIX)) {
+                String propName = key.substring(KAFKA_EXTENSION_PREFIX.length());
+                if (ConsumerConfig.configNames().contains(propName)) {
+                    consumerProperties.put(propName, entry.getValue());
+                } else if (ProducerConfig.configNames().contains(propName)) {
+                    producerProperties.put(propName, entry.getValue());
```
Comment (gmunozfe) [inline kie-server-parent/kie-server-services/kie-server-services-kafka/src/main/java/org/kie/server/services/jbpm/kafka/KafkaServerExtension.java:264]: I think these "ifs" shouldn't be nested, as some properties (`CommonClientConfigs`) belong to both consumer and producer.

**Review by gmunozfe — CHANGES_REQUESTED** [review]
Take into account that some properties have the same name for producer & consumer (CommonClientConfigs) and they should be added to both lists.
Also, a comment related to exception caught when getting the producer and a minor comment for enhancing the test.

File: kie-server-parent/kie-server-services/kie-server-services-kafka/src/main/java/org/kie/server/services/jbpm/kafka/KafkaServerProducer.java (hunk 2 above)
Comment (fjtirado) [reply kie-server-parent/kie-server-services/kie-server-services-kafka/src/main/java/org/kie/server/services/jbpm/kafka/KafkaServerProducer.java:82]: yes, this was caused by a problem in my ide

File: kie-server-parent/kie-server-services/kie-server-services-kafka/src/main/java/org/kie/server/services/jbpm/kafka/KafkaServerExtension.java (hunk 3 above)
Comment (fjtirado) [reply kie-server-parent/kie-server-services/kie-server-services-kafka/src/main/java/org/kie/server/services/jbpm/kafka/KafkaServerExtension.java]: good point, changed

**Review by gmunozfe — APPROVED** [review]
Looks good to me, excellent work @fjtirado !

**Comment by elguardian** [comment]
Jenkins retest this please

**Comment by elguardian** [comment]
Jenkins retest this please

**Pull request merged**